src/plugins/data/server/search/opensearch_search/opensearch_search_strategy.ts

```diff
@@ -92,7 +105,7 @@ export const opensearchSearchStrategyProvider = (
       } catch (e) {
         if (usage) usage.trackError();
 
-        if (dataSource && request.dataSourceId) {
+        if (dataSource?.dataSourceEnabled()) {
```

Comment:
Why do we need to change this condition? It will only check config value? I think previous condition give better coverage.